### PR TITLE
[GCP] Add redis data stream

### LINKED ISSUE #4293: [GCP] Collect metrics for Redis
GCP allows running Redis through Memorystore for Redis.

[There is a set of metrics](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-redis) available trough GCP Cloud Monitoring that we are not currently collecting.

### PR DESCRIPTION
**Pull request by gpop63**
## What does this PR do?

Adds `redis` data stream.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #4293

### CODE REVIEW

File: packages/gcp/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.12.6"
```
Comment (endorama) [inline packages/gcp/changelog.yml:2]:
```suggestion
- version: "2.12.0"
```

File: packages/gcp/data_stream/redis/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
+---
+description: Pipeline for parsing GCP Redis metrics.
+processors:
```
Comment (endorama) [inline packages/gcp/data_stream/redis/elasticsearch/ingest_pipeline/default.yml:3]: All the fields here should have proper suffixes, like mentioned in https://www.elastic.co/guide/en/beats/devguide/current/event-conventions.html

File: packages/gcp/data_stream/redis/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
+- name: gcp.redis
+  description: Google Cloud Redis metrics
+  type: group
+  fields:
```
Comment (endorama) [inline packages/gcp/data_stream/redis/fields/fields.yml:4]: Please add `unit`, `metric_type` and `dimension` where appropriate (see https://github.com/elastic/integrations/issues/4099 for more details)

File: packages/gcp/manifest.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 name: gcp
 title: Google Cloud Platform
-version: "2.11.6"
+version: "2.12.6"
```
Comment (endorama) [inline packages/gcp/manifest.yml:3]:
```suggestion
version: "2.12.0"
```

**Review by endorama — CHANGES_REQUESTED** [review]

**Comment by endorama** [comment]
@gpop63 please fix the CI (needs a `build`)

**Review by endorama — APPROVED** [review]
Apart from the red CI looks good to me!

**Comment by gpop63** [comment]
Tested data stream with new field units etc and works as expected.

File: packages/gcp/data_stream/redis/fields/agent.yml [hunk 5]
```diff
@@ -0,0 +1,198 @@
+- name: cloud
+  title: Cloud
+  group: 2
+  description: Fields related to the cloud or infrastructure the events are coming from.
+  footnote: 'Examples: If Metricbeat is running on an GCP Compute VM and fetches data from its host, the cloud info contains the data about this machine. If Metricbeat runs on a remote machine outside the cloud and fetches data from a service running in the cloud, the field contains cloud data from the machine the service is running on.'
+  type: group
```
Comment (efd6) [inline packages/gcp/data_stream/redis/fields/agent.yml:6]: Some of these fields are duplicates of fields that are defined in ecs.yml.

**Pull request merged**